tests/testthat/test-tab_afun_cfun.R

```diff
@@ -0,0 +1,233 @@
+        split_rows_by("SEX", split_fun = keep_split_levels(c("F", "M"))) %>%
+        add_overall_col("All Patients 2") %>% 
+        analyze(vars = "BMRKR1", afun = n_wrapper_alt_df(ex_adsl))
+    
+    # NB: If you add keep_levels = c("all_X") to add_combo_levels the other 
+    #     column expressions are missing -> Expected!
+    
+    tbl <- lyt %>% build_table(DM, alt_counts_df = ex_adsl)
+    
+    expect_silent(cbind_rtables(tbl, tbl))
+    expect_silent(rbind(tbl, tbl))
+    
+    spl_ctx_cnt <- lapply(seq(8, nrow(tbl), 5), function(x) tbl[x, 2, drop = TRUE])
+    
+    nrow_manual <- lapply(sort(unique(ex_adsl$STRATA1)), function(x) {
+        tmp_strata <- ex_adsl %>% filter(STRATA1 == x, SEX == "F", COUNTRY == "USA")
+        sapply(list(tmp_strata %>% filter(ARM == "A: Drug X"),
+                 tmp_strata %>% filter(ARM == "C: Combination")), nrow)
+    })
+    
+    expect_identical(nrow_manual, spl_ctx_cnt)
+})
+
+test_that("Error localization for missing split variable when done in alt_count_df", {
+    # Error we want to happen
+    afun_tmp <- function(x, .alt_df, ...) mean(x)
+    lyt_col <- basic_table() %>% split_cols_by("ARMCD") %>% analyze("BMRKR1", afun = afun_tmp)
+    lyt_row <- basic_table() %>% split_rows_by("ARMCD") %>% analyze("BMRKR1", afun = afun_tmp)
+    expect_error(lyt_col %>% build_table(ex_adsl, alt_counts_df = DM))
+    expect_error(lyt_row %>% build_table(ex_adsl, alt_counts_df = DM), 
+                 regexp = paste0("Following error encountered in splitting ",
+                 "alt_counts_df:  variable\\(s\\) \\[ARMCD\\] ",
+                 "not present in data. \\(VarLevelSplit\\)"))
+    
+    # Split not requested so no error
+    expect_silent(lyt_row %>% build_table(ex_adsl))
+    expect_silent(lyt_col %>% build_table(ex_adsl))
+    
+    # What if it is not asked by the function?
+    afun_tmp <- function(x, ...) mean(x)
+    lyt_col <- basic_table() %>% 
+        split_cols_by("ARMCD") %>% 
+        analyze("BMRKR1", afun = afun_tmp)
+    lyt_row <- basic_table() %>% 
+        split_rows_by("STRATA1", split_fun = keep_split_levels("A")) %>% 
+        split_rows_by("ARMCD") %>% 
+        analyze("BMRKR1", afun = afun_tmp)
+    
+    # Error on the columns should happen because it is used for counts on column space
+    expect_error(lyt_col %>% build_table(ex_adsl, alt_counts_df = DM))
+    expect_silent(lyt_row %>% build_table(ex_adsl, alt_counts_df = DM))
+})
+
+context("Content functions (cfun)")
+
+test_that(".alt_df appears in cfun but not in afun.", {
+    # Adding STRATA2 col to DM for alt_counts_df col split
+    alt_tmp <- DM %>% left_join(ex_adsl %>% 
+                                    mutate(ID = paste0("S", seq_len(nrow(ex_adsl)))) %>% 
+                                    select(ID, STRATA2))
+    
+    afun_tmp <- function(x, ...) rcell(mean(x), label = "MeAn", format = "xx.x")
+    cfun_tmp <- function(x, .alt_df, labelstr, .N_col, 
```

Comment:
When I print the arguments like `.alt_df` I see a warning:

```
Warning messages:
1: In get(object, envir = currentEnv, inherits = TRUE) :
  restarting interrupted promise evaluation
```

Reply:
I still guess you entered the function from the browser() and it was missing, so probably it was triggering the error across the loop? Or how did you get the error?

Reply:
I put browser() inside `cfun_tmp`.  Here's a better reprex:

```
ADSL <- ex_adsl

ccfun <- function(df, labelstr, .N_col, .alt_df) {
    in_rows(
        rcell(nrow(df) * c(1, 1 / .N_col), format = "xx (xx.xx%)"),
        .labels = labelstr
    )
}

lyt <- basic_table(show_colcounts = TRUE) %>%
    split_rows_by("STRATA1", split_fun = keep_split_levels("A")) %>% 
    summarize_row_groups(cfun = ccfun)  %>%
    analyze("BMRKR1")

build_table(lyt, ADSL) # see warning
build_table(lyt, ADSL, alt_counts_df = DM) # no warning

```

I only see the warning in the case of `build_table` without `alt_counts_df` and the cfun includes `.alt_df`.  It's odd because the warning persists once it pops up and then disappears  :
![image](https://github.com/insightsengineering/rtables/assets/43623360/bae667a1-25f5-4bd3-9f57-37cbcfc2a0fb)

Reply:
```r
> build_table(lyt, ADSL) # see warning
Called from: parent_cfun(df = df, .N_col = .N_col, .something = .something, 
    labelstr = clabelstr, ...)
Browse[1]> .something
Error in (function (df, .N_col, .something, ...)  : 
  argument ".something" is missing, with no default
In addition: Warning messages:
1: In get(object, envir = currentEnv, inherits = TRUE) :
  restarting interrupted promise evaluation
2: restarting interrupted promise evaluation 
Browse[1]> labelstr
[1] "A"
Warning message:
In get(object, envir = currentEnv, inherits = TRUE) :
  restarting interrupted promise evaluation
Browse[1]> Q
```
This happens also on main with .something